### DISCUSSION TITLE
Schmid factor distribution by slip plane/direction

### DISCUSSION THREAD

**Discussion by trudator** [mtex-toolbox/mtex#1705] [Ask Anything] [answered]
Hi folks,

My idea is to find the distribution of Schmid's factor by each slip system after symmetrizing (under the form of (hkl)[uvw]) to find the most and second most activated slip plane/direction among a microstructure after deformation. I found my SF values by using the code below but had no idea how to retrieve the corresponding slip system of each grain. Any idea would be much appreciated. Thanks.
```

sS = slipSystem(Miller(1,-1,1,ebsd('BCC').CS,'uvw'),Miller(0,1,1,ebsd('BCC').CS,'hkl'))
sS = sS.symmetrise;
sSLocal = grains('Martensite BCC').meanOrientation * sS
% compute Schmid factor
sigma = stressTensor.uniaxial(vector3d.Z);
SF = sSLocal.SchmidFactor(-sigma);
% take the maxium along the rows
[SFMax,active] = max(abs(SF),[],2);
% plot the maximum Schmid factor
close all;
figure;
plot(grains('BCC'),SFMax,'micronbar','off','linewidth',2);
```

**Comment by kilir** [comment, marked as answer]
Hi,
in your example, `SF` is number_of_grains-by-slip_systems_in_specimen_coordinates. You find the index of the slip system  with the largest Schmid factor using the line `[SFMax,active] = max(abs(SF),[],2);` and  `sSLocal(active)`.

If you'd like the 2nd, 3rd ...  largest, one possibility would be to use `sort()`, maybe this could work:
```
% sort SF
[~,si]=sort(abs(SF),2,'descend')

% largest SF and cooresponding SSLocal
ind=sub2ind(size(SF),(1:size(SF,1))',si(:,1));
SFMAX=abs(SF(ind))
sSLocalMAX = sSLocal(ind) % sSLocal could also be indexed with si(:,1)

% next smaller value ...
ind=sub2ind(size(SF),(1:size(SF,1))',si(:,3)); % note here's the 3 since the second is just like the first with opposite sign
SF2nd=abs(SF(ind))
sSLocal2nd = sSLocal(ind)
```
Hope that helps.
Cheers,
Rüdiger

**Reply by trudator** [reply]
Awesome, thanks Rudiger for the help. I didn't realize that the sSLocal contains specimen coordinates. 
Quick question: I want to convert back to crystal coordinate under the form of (hkl)[uvw] using `inv(grains('BCC').meanOrientation) * sSLocal` but the results seem wrong. Could you suggest any ideas on this? Thanks.

**Reply by kilir** [reply]
Slip system in crystal  coordinates is simply `sS` in your case.

**Reply by trudator** [reply]
Sorry, I think I misinterpreted my question. My ask is how to convert each of the specimen coordinate in `sSLocal ` to the crystal coordinates. My purpose is to find the crystal coordinate that corresponds to the maximum Schmid factor of each grain.  The `sS` basically only consists of 24 defined slip systems. Thanks.

**Reply by kilir** [reply]
If you convert `sSLocal` to crystal coordinates you get `sS`. You can  index `sS` the same way as  `sSLocal`.

**Reply by trudator** [reply]
Thanks Rudiger, I figured it out.

**Reply by abhi12019** [reply]
Hi,
I am also working on finding out the slip trace analysis. I am not able to figure out, how to plot the active slip planes and direction in individual grains, such as in the BCC phase for a particular grain, <111>{011} this could visualize in the map. Basically I want to create a map to see what slip systems are activated during deformation in the individual grain? Thank you in advance for your help.
